app/services/mentors/add_profile_to_ect.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+module Mentors
+  class AddProfileToECT < BaseService
+    include SchoolCohortDelegator
+
+    def call
+      check_no_mentor_profiles_exist!
+
+      mentor_profile = nil
+
+      ActiveRecord::Base.transaction do
+        mentor_profile = ParticipantProfile::Mentor.create!(
+          teacher_profile: ect_profile.teacher_profile,
+          participant_identity: ect_profile.participant_identity,
+          schedule: Finance::Schedule::ECF.default_for(cohort: school_cohort.cohort),
+          sparsity_uplift: sparsity_uplift?(start_year),
```

Comment:
slight digression, can this change from one year to the next? if so, does this need to be specific to the `SchoolCohort` and not `School`

Reply:
They can change yes, we will get a new list of uplifts every year from policy.  These are now both stored in the `PupilPremium` records attached to the `School` for each year. I _guess_ it might make sense for them to be on `SchoolCohort` but these could be added before any cohort programme decision is made at a given school so there would be nowhere to put them at the point of importing them.